### DISCUSSION TITLE
Types for VercelRequest and VercelResponse serverless functions

### DISCUSSION THREAD

**Discussion by lorypelli** [vercel/community#2176] [Idea]
I can't see them, idk if this is a bug or if they aren't actually avaible, if they aren't avaible, it will be good to add them...

**Comment by devatina11yb** [comment]
You need to install `@vercel/node` and then import the types with `import type { VercelRequest, VercelResponse } from "@vercel/node";`.

**Reply by lorypelli** [reply]
I'm using a javascript file with commonjs, if I use @param it doesn't work

**Reply by devatina11yb** [reply]
Can you please provide an example of what you’re trying to do?

**Reply by lorypelli** [reply]
> Can you please provide an example of what you’re trying to do?

```js
const { VercelRequest, VercelResponse } = require("@vercel/node")
module.exports = async (request, response) => {
     /**
     * @param { VercelRequest } request
     * @param { VercelResponse } response
     */
}
```
but it doesn't work because if I hover over `request` or `response` it says `any`.

**Reply by devatina11yb** [reply]
I believe you can’t `require("@vercel/node")` but you can `import("@vercel/node")` inside JSDoc:
```js
/**
 * @param {import("@vercel/node").VercelRequest} request
 * @param {import("@vercel/node").VercelResponse} response
 */
module.exports = async (request, response) => {
  // do something
};
```

**Reply by lorypelli** [reply]
> ```js
> /**
> * @param {import("@vercel/node").VercelRequest} request
> * @param {import("@vercel/node").VercelResponse} response
> */
> module.exports = async (request, response) => {
>  // do something
> };
> ```

Thanks

**Reply by devatina11yb** [reply]
You’re welcome!

Please mark your question as answered.
